Htmldev Bundle v4

7ochem opened the pull request.

7ochem committed:
Htmldev Bundle v4

erik-zicht commented on README.md:
```diff
@@ -173,22 +173,6 @@ helper function:
-       Validate the provided template _context
-       For rules, see: https://github.com/rakit/validation#available-rules
-    #}
-    {{ _context|validate_context({
```
Heb je hier nog op gezocht in Github? Ik meen me te herinneren dat dit in sommige projecten wel gebruikt wordt.

7ochem replied:
Klopt, vandaag dat ik het nu pas verwijder in deze nieuwe major versie (had het natuurlijk allang veel eerder eruit willen gooien, maar dat kon om die reden niet)

Hangloozz commented:
Wat is samengevat de aanpassing? Ik kan het er niet helemaal uit halen, behalve een flinke aanpassing aan de svg-toolie :)

7ochem commented:
> Wat is samengevat de aanpassing? Ik kan het er niet helemaal uit halen, behalve een flinke aanpassing aan de svg-toolie :)

```
## 4.0.0 - 2022-04-29
### Removed
- Removed support for Symfony 3.4
- Removed support for PHP 7.2 and 7.3
- Rakit/validation package dependency and the `validate_context` Twig filter and the whole ValidateExtension
### Changed
- Replaced use of `Symfony\Bundle\FrameworkBundle\Templating\EngineInterface` with `Twig\Environment`
- Replaces SVG Service deprecated Simple Cache with Cache Adapters
## Fixed
- Fixed several other deprecations
## Added
- Added this CHANGELOG.md
```

😁 

Ik wil een nieuwe major versie maken om bepaalde dingen eruit te kunnen gooien (support voor oudere versies van PHP en Sf) om aanpassingen te kunnen doen die alleen voor Sf 4.4 zijn i.p.v. `if Sf 4.4 doe dit nieuwe ding, else doe de oude manier` wat lelijk is en maintenance lastiger maakt. (Die aanpassing en die SVG tool waren nodig omdat Sf 4.4 een andere manier van cache gebruikt.)

Dus @Hangloozz als je nog dingen hebt op je verlanglijstje voor de Htmldev bundle die breaking zijn, nu is het moment

erik-zicht commented:
Is het moment al aangebroken om hier https://github.com/zicht/htmldev-bundle/blob/release/3.x/src/Resources/views/_base.html.twig#L15  `Fabrique` neer te gaan zetten?

7ochem commented:
> Is het moment al aangebroken om hier https://github.com/zicht/htmldev-bundle/blob/release/3.x/src/Resources/views/_base.html.twig#L15 `Fabrique` neer te gaan zetten?

Ja goed idee. Zal em even aanpassen. Kun je gelijk zien aan de source code welke projecten op Htmldev v4 draaien 😁 

Die header aanpassen in de PHP files (`@copyright Zicht online <https://zicht.nl>`) lijkt me dan weer geen goed idee, want dan krijg je hele nutteloze changesets. Misschien als je het tegenkomt of toch een algehele CS clean up aan het doen bent. Magento had ooit het briljante idee om elk jaar een bugfix versie uit te brengen om alle `(C) 2007-2021` headers te updaten naar `(C) 2007-2022` 🙈

7ochem committed:
Update copyright comment: Changed Zicht to Fabrique

Hangloozz commented:
Euh, goeie ja, ik had het gewoon in de changelog kunnen zien.. dat domme 🤦‍♂️ 

- En hmm, gelijk Tailwind er in hangen? 😄 
- Een toggle-bare zijnavi, zodat componenten ook echt de volle breedte kunnen gebruiken om hun gedrag te checken?
- De naam van het project in de bovenbalk (soms heb ik meerdere htmldev's open staan en dan kun je soms niet snel zien naar welk project je nu kijkt.
- een overzicht van de brand-logos op een manier zoals de iconen dat doen
- de kleurnamen klikbaar en dat ze dan onder je ctrl-v zitten bij klik
- en dat ook gelijk bij de iconen implementeren

Meer niet eigenlijk 😌

7ochem commented:
> * En hmm, gelijk Tailwind er in hangen? :smile:

Die is wel breaking ja 😆 Heb je tijd hiervoor? 😉

Hangloozz commented:
Deze week denk ik niet ivm team-1-perikelen, maa rmisschien volgende week wel.
Of een `htmldev-bundle-v4-tw` uitbrengen zodat ZSS ook nog ondersteund kan worden 😇

7ochem commented:
Ik ben geen issues tegengekomen bij het gebruik van deze nieuwe versie. Dus van mij mag ie door 😁

7ochem committed:
Add Vimeo Psalm and fix errors
